src/edgepi/digital_input/edgepi_digital_input.py

```diff
@@ -12,14 +14,32 @@ def __init__(self):
         # To limit access to input functionality, using composition rather than inheritance
         self.gpio = EdgePiGPIO()
 
-    def digital_input_state(self, pin_name: DinPins = None):
+    def digital_input_state(self, pin_name: Optional[DinPins] = None):
         """
         Read selected GPIO pin
         Args:
             pin_name (DinPins): GpioPin enums
         Return:
             state (bool): corresponding pin state
         """
-        if pin_name is None or pin_name.value not in [pins.value for pins in DinPins]:
+        if pin_name is None or not isinstance(pin_name, DinPins):
             raise InvalidPinName(f'Invalid pin name passed: {pin_name}')
-        return self.gpio.read_pin_state(pin_name.value)
+
+        pin_number = int(pin_name.value[3])
+        return self.gpio.fast_read_din_state(pin_number)
+
+    def digital_input_state_batch(self, pin_names: list[DinPins]) -> list:
+        """
+        Read multiple GPIO pins as digital inputs
+        """
+        if pin_names is None:
+            raise ValueError('pin_names cannot be None')
+        elif pin_names == []:
+            raise ValueError('pin_names cannot be empty')
```

Comment:
Why not use `elif pin_names`?

The PEP8 style guide recommends to use the truthiness check for empty lists: https://peps.python.org/pep-0008/#programming-recommendations

It also might a little more efficient: https://medium.com/swlh/efficiently-checking-for-an-empty-list-in-python-76b76099fbd3

Reply:
> elif pin_names

I usually prefer being more explicit in small cases like showing that pin_names is expected to be a list. However, it seems there is some [significant disagreement](https://stackoverflow.com/questions/53513/how-do-i-check-if-a-list-is-empty) about which way is best, haha.

Compared to the amount of time waiting for IO the check should be negligible, however that's good to know for tight loops!

https://github.com/EdgePi-Cloud/edgepi-python-sdk/pull/423/commits/0b56fb92df5a384b023a311b3cef9dc5b49ccb0f

Reply:
Yeah readability matters too for sure...you could still go with `len(pin_names) == 0` in that case though.